### ISSUE TITLE
Advanced idexing NoneType support

### ISSUE THREAD

**Issue by riccardofelluga** [Lightning-AI/lightning-thunder#187]
## 🐛 Bug

In some edge cases it is possible to encounter a None value as index along with tensor indices. In such cases out  `_advanced_indexing` clang op fails.

### To Reproduce

Repro script:

```python
import torch; import thunder

a = torch.randn(15, 15, 15)
indices  = torch.tensor([1, 2, 3])

def apply_indices(a, b):
    return a[b, b, None]

jfn= thunder.jit(apply_indices)

jfn(a, indices)
```

This fails with the following:

```shell
File /lightning-thunder/thunder/clang/__init__.py:699, in _advanced_indexing(a, key)
    697 if x is Ellipsis:
    698     num_ellipses += 1
--> 699 utils.check(
    700     isinstance(x, (EllipsisType, TensorLike)),
    701     lambda: f"Advanced indexing currently only supports tensors as sequence elements (possibly with a starting ellipsis), but found an object of type {type(x)}",
    702 )
    703 if isinstance(x, TensorLike):
    704     utils.check(
    705         dtypes.is_nonboolean_integer_dtype(x.dtype) and (x.ndim == 1 or x.ndim == 0),
    706         lambda: f"Advanced indexing currently only supports zero or one-dimensional integer tensors, but found a tensor with dtype {x.dtype} and {x.ndim} dimensions",
    707     )

File /lightning-thunder/thunder/core/baseutils.py:103, in check(cond, s, exception_type)
     98 """Helper function for raising an error_type (default: RuntimeError) if a boolean condition fails.
     99 
    100 s is a callable producing a string to avoid string construction if the error check is passed.
    101 """
    102 if not cond:
--> 103     raise exception_type(s())

RuntimeError: Advanced indexing currently only supports tensors as sequence elements (possibly with a starting ellipsis), but found an object of type <class 'NoneType'>
```

### Expected behavior

That's actually a good question, in theory it should create a new axis at the position but sometimes it doesn't, like in this case where the output shape is `(3, 1, 15)`. Corrections about this statement are very welcome.

### Additional context

Such code can be found as part of Mixtral 8x7B.

cc. @mruberry

**Comment by t-vi** [comment]
@k223kim wants to work on this. Thank you!

**Comment by k223kim** [comment]
Hey Team ⚡️, as Tom mentioned, I will be working on this issue. Thanks!

**Comment by riccardofelluga** [comment]
Great to hear @k223kim! Do you already know how do you want to approach it?

**Comment by k223kim** [comment]
Hey @riccardofelluga, thanks for asking! It took a while as I was blocked with a different issue (the detailed `__repr__` PR). I have submitted a draft PR that demonstrates my approach. It'll be awesome if you can take a look and provide some feedback :)

**Comment by k223kim** [comment]
I think this issue can be closed as the merged PR handles it?

**Comment by t-vi** [comment]
Fixed by #710
